zulip_botserver/tests/test_server.py

```diff
@@ -273,6 +274,38 @@ def test_load_lib_modules(self) -> None:
             ).as_posix()
             module = server.load_lib_modules([path])[path]
 
+    @mock.patch("zulip_botserver.server.app")
+    @mock.patch("sys.argv", ["zulip-botserver", "--config-file", "/foo/bar/baz.conf"])
+    def test_load_from_registry(self, mock_app: mock.Mock) -> None:
+        packaged_bot_module = mock.MagicMock(__version__="1.0.0", __file__="asd")
+        packaged_bot_entrypoint = metadata.EntryPoint(
+            "packaged_bot", "module_name", "zulip_bots.registry"
+        )
+        bots_config = {
+            "packaged_bot": {
+                "email": "packaged-bot@zulip.com",
+                "key": "value",
+                "site": "http://localhost",
+                "token": "abcd1234",
+            }
+        }
+
+        with mock.patch("zulip_botserver.server.read_config_file", return_value=bots_config):
+            with mock.patch("zulip_botserver.server.lib.ExternalBotHandler", new=mock.Mock()):
+                with mock.patch(
+                    "zulip_bots.finder.metadata.EntryPoint.load",
+                    return_value=packaged_bot_module,
+                ):
+                    with mock.patch(
+                        "zulip_bots.finder.metadata.entry_points",
+                        return_value=(packaged_bot_entrypoint,),
+                    ):
```

Comment:
You don’t need to nest `with` blocks—just use [commas](https://docs.python.org/3/reference/compound_stmts.html#the-with-statement).